pkg/utils/math_test.go

```diff
@@ -33,3 +36,47 @@ func TestOffsetRightNoOverflow(t *testing.T) {
 	assert.Equal(t, uint64(18446744073709551603), res)
 	assert.False(t, isOverflow)
 }
+
+func TestFeltDivRem(t *testing.T) {
+	type testCase struct {
+		a fp.Element
+		b fp.Element
+	}
+	tests := []testCase{
+		{fp.NewElement(0), fp.NewElement(1)},
+		{fp.NewElement(10), fp.NewElement(2)},
+		{fp.NewElement(2), fp.NewElement(10)},
+		{fp.NewElement(10), fp.NewElement(9)},
+		{fp.NewElement(9), fp.NewElement(10)},
+	}
+
+	// Add some randomized values to the test.
+	// I'll use a fixed seed to avoid flaky results.
+	rng := rand.New(rand.NewSource(83285239))
+	for i := 0; i < 300; i++ {
+		tests = append(tests, testCase{
+			fp.NewElement(uint64(rng.Uint32())),
+			fp.NewElement(uint64(rng.Uint32()) + 1),
+		})
+	}
+
+	for i, test := range tests {
+		haveDiv, haveRem := FeltDivRem(&test.a, &test.b)
+
+		var a big.Int
+		var b big.Int
+		var div big.Int
+		var rem big.Int
+		test.a.BigInt(&a)
+		test.b.BigInt(&b)
+		div.DivMod(&a, &b, &rem)
+
+		wantDiv := new(fp.Element).SetBigInt(&div)
+		wantRem := new(fp.Element).SetBigInt(&rem)
+
```

Comment:
Even though I like the idea of checking random values, I don't think this is very helpful in this case, as the way to calculate the expected result is the same code you want to test. I rather just go with the fixed cases